### ISSUE TITLE
If step is decimal, slider does not show ceil

### ISSUE THREAD

**Issue by Etiake** [angular-slider/ngx-slider#128]
Goodmorning,
I want to thank the author(s) for this beautiful slider, I found it very useful! But... I noticed this issue with decimal step in Slider: if the step is decimal, the slider does not show the ceil value. To see the issue, you can use this stackblitz:
[https://stackblitz.com/edit/ng5-slider-simple-slider-example?file=src%2Fapp%2Fapp.component.ts ](
)
with this code:

```
export class AppComponent  {
  value: number = 5;
  options: Options = {
    floor: 4,
    ceil:6,
    step: 0.1,
    showTicks: true,
    showTicksValues: true
  };
}
```

Is it a configuration issue? Or is it a bug? Thank you if you can help me to understand.

**Comment by Etiake** [comment]
As I can see, decimal values are not displayed even if showTicksValues is true. Can this be related to the missing ceil value?

**Comment by kj9933** [comment]
Just Use like this
options: Options = {
    floor: -10,
    ceil: 10,
    step: 0.1
  };
Its working for me.

**Comment by piotrdz** [comment]
There were two problems here:
1) decimal rounding in calculations prevented the ticks from displaying correctly - the final tick on the ceil value was missing
2) the decision on whether to display a tick value at a given position was controlled by `tickValueStep` option with a default value of 1 - this led to a confusing behaviour if you didn't set it to something else.

I fixed the calculations in 1) so that ticks should display correctly and for 2) I changed the default behaviour of `tickValueStep` - it is now optional and will fall back to `tickStep` or `step` if they are defined.

Merged to master, awaiting next release.
